cfme/cloud/security_group.py

```diff
@@ -1,20 +1,268 @@
+)
+from widgetastic.widget import View, Select
+from widgetastic_patternfly import Dropdown, Button
+from cfme.base.ui import BaseLoggedInPage
+from cfme.exceptions import ItemNotFound, SecurityGroupsNotFound
+from cfme.web_ui import match_location
+from utils.appliance.implementations.ui import navigate_to, navigator, CFMENavigateStep
+from utils.appliance import Navigatable
+from utils.wait import wait_for
 
 
-:var list_page: A :py:class:`cfme.web_ui.Region` object describing elements on the list page.
-:var details_page: A :py:class:`cfme.web_ui.Region` object describing elements on the detail page.
-"""
+class SecurityGroupsToolbar(View):
+    configuration = Dropdown('Configuration')
+    policy = Dropdown('Policy')
+    download = Dropdown('Download')
 
-from cfme.web_ui import Region, SplitTable
+    view_selector = View.nested(ItemsToolBarViewSelector)
 
 
-# Page specific locators
-list_page = Region(
-    locators={
-        'security_group_table': SplitTable(header_data=('//div[@class="xhdr"]/table/tbody', 1),
-            body_data=('//div[@class="objbox"]/table/tbody', 1))
-    },
-    title='Security Groups')
+class SecurityGroupsDetailsToolbar(View):
+    configuration = Dropdown('Configuration')
+    policy = Dropdown('Policy')
+    download = Button(title='Download summary in PDF format')
 
 
-details_page = Region(infoblock_type='detail')
+class SecurityGroupsDetailsAccordion(View):
+    @View.nested
+    class properties(Accordion):  # noqa
+        tree = ManageIQTree()
+
+    @View.nested
+    class relationships(Accordion):  # noqa
+        tree = ManageIQTree()
+
+
+class SecurityGroupsQuadIconEntity(BaseQuadIconEntity):
+    pass
+
+
+class SecurityGroupsTileIconEntity(BaseTileIconEntity):
+    quad_icon = ParametrizedView.nested(SecurityGroupsQuadIconEntity)
+
+
+class SecurityGroupsListEntity(BaseListEntity):
+    pass
+
+
+class NonJSSecurityGroupsEntity(NonJSBaseEntity):
+    quad_entity = SecurityGroupsQuadIconEntity
+    list_entity = SecurityGroupsListEntity
+    tile_entity = SecurityGroupsTileIconEntity
+
+
+def SecurityGroupsEntity():  # noqa
+    """Temporary wrapper for Volume Entity during transition to JS based Entity """
+    return VersionPick({
+        Version.lowest(): NonJSSecurityGroupsEntity,
+        '5.9': JSBaseEntity,
+    })
+
+
+class SecurityGroupsEntities(BaseEntitiesView):
+    @property
+    def entity_class(self):
+        return SecurityGroupsEntity().pick(self.browser.product_version)
+
+
+class SecurityGroupsDetailsEntities(View):
+    breadcrumb = BreadCrumb()
+    title = Text('//div[@id="main-content"]//h1')
+    table = Table("//div[@id='list_grid']//table")
+    properties = SummaryTable(title='Properties')
+    relationships = SummaryTable(title='Relationships')
+    smart_management = SummaryTable(title='Smart Management')
+    firewall_rules = SummaryTable(title="Firewall Rules")
+
+
+class SecurityGroupsAddEntities(View):
```

Comment:
There are some duplications here between entities and form, I think entities should only include breadcrumb and title widgets, removing network_manager and cloud_tenant because they're in form.

Reply:
Ok removed.